tests/queries/tests.py

```diff
@@ -1173,6 +1173,23 @@ def test_common_mixed_case_foreign_keys(self):
         )
         self.assertTrue(qs.first())
 
+    def test_excluded_intermediary_m2m_table_joined(self):
+        # Three JOINs on "note" column.
```

Comment:
I'm not sure what this comment means.

The first query generates

```sql
SELECT
    "queries_note"."id",
    "queries_note"."note",
    "queries_note"."misc",
    "queries_note"."tag_id"
FROM
    "queries_note"
WHERE
    NOT ("queries_note"."tag_id" IN (
            SELECT
                U2. "tag_id"
            FROM
                "queries_annotation" U2
            WHERE
                U2. "name" = "queries_note"."note")
            AND "queries_note"."tag_id" IS NOT NULL)
    ORDER BY
        "queries_note"."note" ASC
```

And the second

```sql
SELECT
    "queries_note"."id",
    "queries_note"."note",
    "queries_note"."misc",
    "queries_note"."tag_id"
FROM
    "queries_note"
    INNER JOIN "queries_tag" ON ("queries_note"."tag_id" = "queries_tag"."id")
    INNER JOIN "queries_annotation" ON ("queries_tag"."id" = "queries_annotation"."tag_id")
    INNER JOIN "queries_annotation" T4 ON ("queries_tag"."id" = T4. "tag_id")
WHERE ("queries_annotation"."name" = 'a1'
    AND T4. "name" = 'a1'
    AND NOT ("queries_note"."tag_id" IN (
            SELECT
                U2. "tag_id"
            FROM
                "queries_annotation" U2
            WHERE
                U2. "name" = "queries_note"."note")
            AND "queries_note"."tag_id" IS NOT NULL))
ORDER BY
    "queries_note"."note" ASC
```

Reply:
I removed comment.